controller/src/main/java/cern/jarrace/controller/io/JarWriter.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * © Copyright 2016 CERN. This software is distributed under the terms of the Apache License Version 2.0, copied
+ * verbatim in the file “COPYING”. In applying this licence, CERN does not waive the privileges and immunities granted
+ * to it by virtue of its status as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+package cern.jarrace.controller.io;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.InvalidPropertyException;
+import org.springframework.beans.factory.annotation.Value;
+
+import javax.annotation.PostConstruct;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.regex.Pattern;
+
+/**
+ * Class responsible of writing the deployment file to the file system
+ *
+ * @author tiagomr
+ */
+public class JarWriter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JarWriter.class);
+    private static final String FILE_EXTENSION = "jar";
+
+    @Value("${io.deploymentdir}")
```

Comment:
There is no default, if the user does not set anything, an exception arrises saying that the placeholder could not be resolved